### ISSUE TITLE
Is auditor using the database during a cache:clear?

### ISSUE THREAD

**Issue by parijke** [DamienHarper/auditor#11]
I upgraded yesterday to 4.0.1. I use a docker environment, but also have a console binary on my host system. Now since yesterday I experience that a bin/console cache:clear is ending in a PDO exception (which is correct since the host environment doesn't have one), but that wasn't the case before.

I think Auditor is doing something with the database connection during cache warmup? Is this correct and expected behaviour  or did I do something wrong?

```
Exception trace:
  at /home/paul/www/BO3/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:37
 PDO->__construct() at /home/paul/www/BO3/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOConnection.php:37
 Doctrine\DBAL\Driver\PDOConnection->__construct() at /home/paul/www/BO3/vendor/doctrine/dbal/lib/Doctrine/DBAL/Driver/PDOMySql/Driver.php:24
 Doctrine\DBAL\Driver\PDOMySql\Driver->connect() at /home/paul/www/BO3/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:360
 Doctrine\DBAL\Connection->connect() at /home/paul/www/BO3/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:425
 Doctrine\DBAL\Connection->getDatabasePlatformVersion() at /home/paul/www/BO3/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:385
 Doctrine\DBAL\Connection->detectDatabasePlatform() at /home/paul/www/BO3/vendor/doctrine/dbal/lib/Doctrine/DBAL/Connection.php:328
 Doctrine\DBAL\Connection->getDatabasePlatform() at /home/paul/www/BO3/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:801
 Doctrine\ORM\Mapping\ClassMetadataFactory->getTargetPlatform() at /home/paul/www/BO3/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:644
 Doctrine\ORM\Mapping\ClassMetadataFactory->completeIdGeneratorMapping() at /home/paul/www/BO3/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:166
 Doctrine\ORM\Mapping\ClassMetadataFactory->doLoadMetadata() at /home/paul/www/BO3/vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php:306
 Doctrine\Persistence\Mapping\AbstractClassMetadataFactory->loadMetadata() at /home/paul/www/BO3/vendor/doctrine/orm/lib/Doctrine/ORM/Mapping/ClassMetadataFactory.php:82
 Doctrine\ORM\Mapping\ClassMetadataFactory->loadMetadata() at /home/paul/www/BO3/vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php:184
 Doctrine\Persistence\Mapping\AbstractClassMetadataFactory->getMetadataFor() at /home/paul/www/BO3/vendor/doctrine/persistence/lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php:90
 Doctrine\Persistence\Mapping\AbstractClassMetadataFactory->getAllMetadata() at /home/paul/www/BO3/vendor/damienharper/auditor/src/Provider/Doctrine/Auditing/Annotation/AnnotationLoader.php:32
 DH\Auditor\Provider\Doctrine\Auditing\Annotation\AnnotationLoader->load() at /home/paul/www/BO3/vendor/damienharper/auditor/src/Provider/Doctrine/DoctrineProvider.php:236
 DH\Auditor\Provider\Doctrine\DoctrineProvider->loadAnnotations() at /home/paul/www/BO3/vendor/damienharper/auditor/src/Provider/Doctrine/DoctrineProvider.php:45
 DH\Auditor\Provider\Doctrine\DoctrineProvider->registerAuditingService() at /home/paul/www/BO3/var/cache/de_/ContainerJBNVfRG/App_KernelDevDebugContainer.php:1518
 ContainerJBNVfRG\App_KernelDevDebugContainer->getDoctrineProviderService() at /home/paul/www/BO3/var/cache/de_/ContainerJBNVfRG/App_KernelDevDebugContainer.php:920
 ContainerJBNVfRG\App_KernelDevDebugContainer->getDoctrine_Dbal_DefaultConnectionService() at /home/paul/www/BO3/var/cache/de_/ContainerJBNVfRG/App_KernelDevDebugContainer.php:933
 ContainerJBNVfRG\App_KernelDevDebugContainer->getDoctrine_Orm_DefaultEntityManagerService() at /home/paul/www/BO3/var/cache/de_/ContainerJBNVfRG/App_KernelDevDebugContainer.php:2310
 ContainerJBNVfRG\App_KernelDevDebugContainer->getValidator_BuilderService() at /home/paul/www/BO3/var/cache/de_/ContainerJBNVfRG/getValidator_Mapping_CacheWarmerService.php:24
 ContainerJBNVfRG\getValidator_Mapping_CacheWarmerService::do() at /home/paul/www/BO3/var/cache/de_/ContainerJBNVfRG/App_KernelDevDebugContainer.php:597
 ContainerJBNVfRG\App_KernelDevDebugContainer->load() at /home/paul/www/BO3/var/cache/de_/ContainerJBNVfRG/getCacheWarmerService.php:25
 ContainerJBNVfRG\getCacheWarmerService::ContainerJBNVfRG\{closure}() at /home/paul/www/BO3/vendor/symfony/http-kernel/CacheWarmer/CacheWarmerAggregate.php:90
 Symfony\Component\HttpKernel\CacheWarmer\CacheWarmerAggregate->warmUp() at /home/paul/www/BO3/vendor/symfony/http-kernel/Kernel.php:580
 Symfony\Component\HttpKernel\Kernel->initializeContainer() at /home/paul/www/BO3/vendor/symfony/http-kernel/Kernel.php:780
 Symfony\Component\HttpKernel\Kernel->preBoot() at /home/paul/www/BO3/vendor/symfony/http-kernel/Kernel.php:121
 Symfony\Component\HttpKernel\Kernel->boot() at /home/paul/www/BO3/vendor/symfony/http-kernel/Kernel.php:139
 Symfony\Component\HttpKernel\Kernel->reboot() at /home/paul/www/BO3/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:224
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->warmup() at /home/paul/www/BO3/vendor/symfony/framework-bundle/Command/CacheClearCommand.php:148
 Symfony\Bundle\FrameworkBundle\Command\CacheClearCommand->execute() at /home/paul/www/BO3/vendor/symfony/console/Command/Command.php:255
 Symfony\Component\Console\Command\Command->run() at /home/paul/www/BO3/vendor/symfony/console/Application.php:989
 Symfony\Component\Console\Application->doRunCommand() at /home/paul/www/BO3/vendor/symfony/framework-bundle/Console/Application.php:96
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at /home/paul/www/BO3/vendor/symfony/console/Application.php:290
 Symfony\Component\Console\Application->doRun() at /home/paul/www/BO3/vendor/symfony/framework-bundle/Console/Application.php:82
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at /home/paul/www/BO3/vendor/symfony/console/Application.php:166
 Symfony\Component\Console\Application->run() at /home/paul/www/BO3/bin/console:42
```

**Comment by DamienHarper** [comment]
@parijke Hum that's not expected so I need to do some investigation here. I'll keep you updated

**Comment by parijke** [comment]
If I can be of any help let me know

**Comment by DamienHarper** [comment]
@parijke After a bit of investigation, finally this is an expected behavior because when warming up the cache, auditor is setup and one of the setup process steps is dedicated to registering auditing and storage services. When registering auditing services, annotations are loaded to build/complete the auditable entities inventory. Loading annotations requires accessing entity metadata from the entity manager (bound to the auditing service) which in turn fires a connection.

I'll try to implement some kind of annotation lazy loading in the future but I don't have any ETA for now.

**Comment by parijke** [comment]
Thanks Damien, for explaining. I now know that is is expected and I have to change my CI chain (I build without a database before). It is as from 4.0.0, the 3.x of dcotrine-audit-bundle did not have this.

**Comment by DamienHarper** [comment]
@parijke this should be fixed since `1.2.0` (PR #30) so I close this issue. Feel free to reopen it if needed.
